frontend/app/controllers/login.js

```diff
@@ -3,12 +3,11 @@ import Ember from 'ember';
 export default  Ember.Controller.extend({
   session: Ember.inject.service('session'),
 
-  actions:{
-    authenticate: function authenticate(provider){
-      this.get('session').authenticate('authenticator:firebase-simple-auth',{
-        provider: provider,
-        username: this.get('username'),
-        password: this.get('password')
+  actions: {
+    authenticate() {
+      let { username, password } = this.getProperties('username', 'password');
+      this.get('session').authenticate('authenticator:oauth2', username, password).catch((reason) => {
```

Comment:
Ah a fat arrow, at some point we'll use them more especially in the backend. Have a read of how to use them perhaps.